tests/unit/disco/components/TestAddon.js

```diff
@@ -411,6 +420,29 @@ describe(__filename, () => {
       });
     });
 
+    it('calls getClientCompatibility to determine the compatibility', () => {
+      const _getClientCompatibility = sinon.mock().returns({
+        compatible: true,
+      });
+      const addon = fakeAddon;
+      const clientApp = CLIENT_APP_FIREFOX;
+      const userAgent = sampleUserAgent;
+      const userAgentInfo = UAParser(userAgent);
+      dispatchClientMetadata({ clientApp, store, userAgent });
+      render({
+        _getClientCompatibility,
+        addon,
+      });
+
+      sinon.assert.calledWith(_getClientCompatibility, {
+        addon: createInternalAddon(addon),
+        clientApp,
+        currentVersion: createInternalVersion(fakeVersion),
+        // userAgentInfo stored in state.api only includes browser and os.
+        userAgentInfo: { browser: userAgentInfo.browser, os: userAgentInfo.os },
```

Comment:
Recreating this object from scratch is fragile. I think it would be fine to just compare it to `root.instance().props.userAgentInfo` because we have Flow type safety to help cover that prop.